pkg/osquery/runtime/runtime.go

```diff
@@ -181,6 +182,12 @@ func (opts *osqueryOptions) createOsquerydCommand(osquerydBinary string, paths *
 		cmd.Stderr = opts.stderr
 	}
 
+	// Apply user-provided flags last so that they can override any other flags
```

Comment:
I am _very_ leery of this footgun, I spend some time debugging footguns.

What settings do you want to override? I probably feel most protective of:
```
		fmt.Sprintf("--pidfile=%s", paths.pidfilePath),
		fmt.Sprintf("--database_path=%s", paths.databasePath),
		fmt.Sprintf("--extensions_socket=%s", paths.extensionSocketPath),
		fmt.Sprintf("--extensions_autoload=%s", paths.extensionAutoloadPath),
		"--extensions_timeout=10",
		fmt.Sprintf("--config_plugin=%s", opts.configPluginFlag),
```

Though I'm not totally sure on config_plugin

Reply:
I can drop those in after these and document that those flags cannot be overridden. If it's ever an issue for someone, they can bring it up and propose a solution.